ccmbioinfo/lfs
# CARDS-2571: Reference Questions do not work inside Conditional Sections

- Update reference question initialization code to check for unlinked reference questions any time a form is edited, not just when a form is created
  - Search for any reference questions on the form that do not have an answer
  - Check if the missing questions have an existing answer section. If so, generate an answer
  - Try to find a source answer for any existing or newly created unanswered reference answers
    - Prioritize pulling source answers from forms that are already referenced by other reference answers in this form for consistency
- Add test case in ReferenceTestCopied form

To test:
1. Launch with test forms
2. Generate a couple instances of ReferenceTestUser forms with the last form having unique text answers
3. Generate a ReferenceTestCopied form.
  a. Make sure the referenced text answers are from the most recent User form
  b. Do not fill out the `Should the conditional section be satisfied` answer
4. Generate at least 1 more instance of ReferenceTestUser forms with different text answers
5. Edit the existing ReferenceTestCopied form and set `Should the conditional section be satisfied` to true
6. Save and close the form
7. Open the Copied form in view mode. Ensure the last section is present and contains the unique text answer from step 2.

Previous (incorrect) behaviour:
- In step 7 the last section would be present but would never contain an answer
